File: stack/stack.js
class Stack {
    constructor() {
        this.size = 0;
        this.stack = [];
    }

    push(val) {
        this.stack.push(val);
        this.size++
    }

    pop() {
        if(this.size === 0) return

        this.stack.pop();
        this.size--
    }

    isEmpty() {
        return this.size === 0
    }

    print() {
        for(let i = this.size - 1; i >= 0; i--)
            console.log(this.stack[i])
    }

}

let s = new Stack()
s.push(1)
s.push(2)
s.push(3)
s.pop()
s.print()

File: stack/stack.cpp
#include "stack.h"

#include <iostream>

Stack::Stack() {
    _size = 0;
}

void Stack::push(int val) {
    _stack.push_back(val);
    _size++;
}

int Stack::pop() {
    if(isEmpty()) return -1;

    int ele = _stack.back();

    _stack.pop_back();
    _size--;

    return ele;
}

bool Stack::isEmpty() {
    return _size == 0;
}

void Stack::print() {
    for(int i = _size - 1; i >= 0; i--) {
        std::cout << _stack[i] << " ";
    }
    std::cout << std::endl;
}